### DISCUSSION TITLE
Question about m4b conversion and iTunes

### DISCUSSION THREAD

**Discussion by pfh-personal** [audiamus/AaxAudioConverter#61] [Q&A] [answered]
In your (excellent!) documentation, you talk about the issues with playing large m4b files in iTunes, and give instructions for how to configure AAX converter to create large, single-file m4b files that will play correctly.

I've had some previous experience with this issue, and my understanding was that you have to adjust the sample rate during conversion based on the audio length. The size limits I've used before are:

   Sample rate of 8.0 kHz = 74.6 hrs max
   Sample rate of 12.0 kHz = 49.7 hrs max
   Sample rate of 16.0 kHz = 37.3 hrs max
   Sample rate of 22.05 kHz = 27.1 hrs max
   Sample rate of 24.0 kHz = 24.9 hrs max
   Sample rate of 32.0 kHz = 18.6 hrs max
   Sample rate of 44.1 kHz = 13.5 hrs max
   Sample rate of 48.0 kHz = 12.4 hrs max

However, after following the instructions in the documentation for creating large m4b files using AAX Converter, I found the sample rate of the single m4b file was 44.1 kHz regardless of the length of the audio book.

Specifically, I enabled "Intermediate copy for single-file mode", and "Fix AAC encoding for 44 kHz" to "In MP4 single-file mode with intermed. copy".

Have I misunderstood the problem, and sample rate is not relevant? Or have a misconfigured something in the utility that prevents reduction of the sample rate?

**Comment by audiamus** [comment, marked as answer]
The intermediate copy is the result of experimenting with FFmpeg, carried out by a very helpful user.

We see the phenomenon but treat it as a black box. I would guess the problem could be a tag, or an "atom" in the meta data lingo, that has an insufficient field width. That could result in an overflow for very long files, where it is uncertain whether long means duration or file size or both. (The latter would make it dependent on the bit rate.)

So, our workaround was to strip off all meta data and create it anew. We do that by extracting the pure AAC audio stream out of the MP4 container (AAX is MP4). Then we take this AAC stream and put it inside a new MP4 container. That is all done with FFmpeg.

The 44.1 kHz issue is a different atom that's going wrong. But in this case, FFmpeg would balk and refuse to extract the AAC stream.
See also here: [https://github.com/KrumpetPirate/AAXtoMP3/issues/116](https://github.com/KrumpetPirate/AAXtoMP3/issues/116)
It's solved by patching the file before feeding it to FFmpeg.

**Reply by pfh-personal** [reply]
Thanks very much for the detailed explanation. I guess my previous experiences with reducing the sample rate worked for reasons that had little to do with the actual problem!

I'll have to search for a tool that converts mp3s to m4bs by recreating the metadata. I have a lot of m4bs that have reduced sample rates because of their size, because of my misunderstanding of the iTunes/iOS problem.
